pelican/server.py

```diff
@@ -3,20 +3,45 @@
 
 import logging
 import os
+import posixpath
 import sys
 
 try:
     from magic import from_file as magic_from_file
 except ImportError:
     magic_from_file = None
 
+from six.moves import BaseHTTPServer
 from six.moves import SimpleHTTPServer as srvmod
-from six.moves import socketserver
+from six.moves import urllib
 
 
 class ComplexHTTPRequestHandler(srvmod.SimpleHTTPRequestHandler):
     SUFFIXES = ['', '.html', '/index.html']
 
+    def translate_path(self, path):
+        # abandon query parameters
+        path = path.split('?', 1)[0]
+        path = path.split('#', 1)[0]
+        # Don't forget explicit trailing slash when normalizing. Issue17324
+        trailing_slash = path.rstrip().endswith('/')
+        try:
+            path = urllib.parse.unquote(path, errors='surrogatepass')
```

Comment:
This code snippet comes directly from the Python 3.5 source code which explains that it doesn't work with Python 2.7. I guess an easy fix would be to just take the same snippet from the Python 2.7 sources instead because that code should work with Python 3.5.